tests/ttnn/unit_tests/operations/test_silu.py

```diff
@@ -0,0 +1,159 @@
+                )
+            }
+        )
+    else:
+        raise ValueError("Invalid ncores")
+
+
+@skip_for_wormhole_b0()
+@pytest.mark.parametrize(
+    "batch_size, input_channels, input_height, input_width, ncores, shard_grid, shard_strategy, shard_orientation",
+    (
+        (2, 320, 64, 64, 40, (8, 5), ttnn.ShardStrategy.BLOCK, ttnn.ShardOrientation.COLUMN_MAJOR),
+        (8, 256, 56, 56, 98, (12, 9), ttnn.ShardStrategy.HEIGHT, ttnn.ShardOrientation.ROW_MAJOR),
+        # (8, 512, 28, 28, 80, (10, 8), ttnn.ShardStrategy.BLOCK, ttnn.ShardOrientation.COLUMN_MAJOR), ##hang
+        (8, 1024, 14, 14, 56, (7, 8), ttnn.ShardStrategy.BLOCK, ttnn.ShardOrientation.COLUMN_MAJOR),
+        (16, 256, 56, 56, 98, (12, 9), ttnn.ShardStrategy.HEIGHT, ttnn.ShardOrientation.ROW_MAJOR),
+        # (16, 512, 28, 28, 88, (11, 8), ttnn.ShardStrategy.BLOCK, ttnn.ShardOrientation.COLUMN_MAJOR),  #HANG
+        # (16, 1024, 14, 14, 72, (9, 8), ttnn.ShardStrategy.BLOCK, ttnn.ShardOrientation.COLUMN_MAJOR),   #HANG
+    ),
+)
+@pytest.mark.parametrize("op", ["silu", "relu"])
+def test_sd_elt(
+    device,
+    batch_size,
+    input_channels,
+    input_height,
+    input_width,
+    shard_grid,
+    ncores,
+    shard_strategy,
+    shard_orientation,
+    op,
+):
+    ## input shape is N C H W
+    input_shape = [batch_size, input_channels, input_height, input_width]
+    torch.manual_seed(0)
+    input = torch.rand(input_shape, dtype=torch.bfloat16) * 2 - 1
+
+    ## permute to N H W C, which is what the upsample op expects
+    tt_input = input.permute(0, 2, 3, 1)
+    tt_input = tt_input.reshape(1, 1, batch_size * input_height * input_width, input_channels)
+    input_tensor = ttnn.from_torch(
+        tt_input, device=device, memory_config=ttnn.L1_MEMORY_CONFIG, layout=ttnn.TILE_LAYOUT
+    )
+    interleaved_mem_config = ttnn.MemoryConfig(ttl.tensor.TensorMemoryLayout.INTERLEAVED, ttl.tensor.BufferType.L1)
+    input_tensor = ttnn.to_memory_config(input_tensor, interleaved_mem_config)
+    # input_shape = [1, 1, _nearest_y(batch_size * input_height * input_width, 32), input_channels]
+    input_2d_height = input_tensor.get_legacy_shape()[2]
+    input_2d_width = input_tensor.get_legacy_shape()[3]
+    logger.debug(f"input_2d_height={input_2d_height} and input_2d_width={input_2d_width}")
+    ## input shard
+    if shard_strategy == ttnn.ShardStrategy.BLOCK:
+        input_2d_height_padded = _nearest_y(input_2d_height, shard_grid[0] * 32)
+        shard_height = math.ceil(input_2d_height_padded / shard_grid[0])
+        shard_width = math.ceil(input_2d_width / shard_grid[1])
+        # shard_orientation = ttnn.ShardOrientation.COLUMN_MAJOR
+        shard_orientation = ttnn.experimental.tensor.ShardOrientation.COL_MAJOR
+        tensor_memory_layout = ttnn.types.TensorMemoryLayout.BLOCK_SHARDED
+    elif shard_strategy == ttnn.ShardStrategy.HEIGHT:
+        shard_height = math.ceil(input_2d_height / ncores)
```

Comment:
We need the height padding to 32 for height sharding too. Basically what you did for block sharding.